### ISSUE TITLE
Failures when Dart updates

### ISSUE THREAD

**Issue by drpond1** [dart-lang/sdk#35203]
When a new version of  dart appears eventually my code will not run -- some kind of script error on startup, before it gets to my code.  (Unfortunately, I did not save the error.)
    This is resolved by updating the dartsdk, getting/updating dependencies and rebuilding.
It seems to me I should still be able to run with the old system.    If not, there should at least be a  message that a dartsdk update is required. 

 This is the second time this has happened in the last few months.      

I use IntelliJ Community Edition.

**Comment by alexander-doroshko** [comment]
@vsmenon, I'm not sure `area-intellij` is the right label here.
Without error details the issue doesn't seem to be actionable at all, so maybe just close?

**Comment by drpond1** [comment]
I believe this was addressed in a recent related issue.   The problem is
that while I am working on a project,   there can be an update from Dart
which when a build happens causes builds to fail.     I think this is new
with Dart 2.0  because of the recommended

build_runner: ^0.10.0
build_web_compilers: ^0.4.


My situation is that I may keep an older app running,

because I want to copy "user" data from the old/flawed app into the new one.


New builds may fail if there is a Dart update between the time the older app was

started and the new build was attempted.  (There cannot be another running

app.)

Because of the build failure, new code executes with a missing file error.


***

The failure is intended by the Dart design and the

shown pub spec is still recommended  so the issue was closed.

***


The residual problem is that the failures I encountered are unexpected

 and recognition of how to recover was not immediately obvious to me.


If this analysis of the problem is correct, the simplest fix is a small

change in documentation.

However, my situation might be sufficiently rare that no change should

be made.
